### ISSUE TITLE
Error type is no longer Sync in Linux

### ISSUE THREAD

**Issue by belak** [kangalioo/launchy#6]
The following program fails to compile against the lp-mini branch:

```rust
use launchy::prelude::*;
use launchy::mini::{Input, Output, Color, Button};

fn main() -> anyhow::Result<()> {
    let mut output = Output::guess()?;

    for i in 0..4 {
        for j in 0..4 {
            output.light(Button::GridButton { x: i, y: j }, Color::new(i, j))?;
        }
    }

    Ok(())
}
```

Here is an excerpt of the compilation error:

```   Compiling lp-control v0.1.0 (/home/belak/Code/lp-control)                                                           
error[E0277]: `*mut alsa_sys::_snd_seq` cannot be shared between threads safely                                        
 --> src/main.rs:9:39                                                                                                  
  |                                                                                                                    
9 |     let input = Input::guess_polling()?;                                                                           
  |                                       ^ `*mut alsa_sys::_snd_seq` cannot be shared between threads safely          
  |                                                                                                                    
  = help: within `MidiError`, the trait `Sync` is not implemented for `*mut alsa_sys::_snd_seq`                        
  = note: required because it appears within the type `alsa::seq::Seq`                                                 
  = note: required because it appears within the type `Option<alsa::seq::Seq>`                                         
  = note: required because it appears within the type `midir::backend::alsa::MidiInput`                                
  = note: required because it appears within the type `midir::common::MidiInput`                                       
  = note: required because it appears within the type `midir::errors::ConnectError<midir::common::MidiInput>`          
  = note: required because it appears within the type `MidiError`                                                      
  = note: required because of the requirements on the impl of `From<MidiError>` for `anyhow::Error`                    
  = note: required by `from`                                                                                           
                                                                                                                       
error[E0277]: `Cell<bool>` cannot be shared between threads safely                                                     
 --> src/main.rs:9:39                                                                                                  
  |                                                                                                                    
9 |     let input = Input::guess_polling()?;                                                                           
  |                                       ^ `Cell<bool>` cannot be shared between threads safely                       
  |                                                                                                                    
  = help: within `MidiError`, the trait `Sync` is not implemented for `Cell<bool>`                                     
  = note: required because it appears within the type `alsa::seq::Seq`                                                 
  = note: required because it appears within the type `Option<alsa::seq::Seq>`                                         
  = note: required because it appears within the type `midir::backend::alsa::MidiInput`                                
  = note: required because it appears within the type `midir::common::MidiInput`                                       
  = note: required because it appears within the type `midir::errors::ConnectError<midir::common::MidiInput>`          
  = note: required because it appears within the type `MidiError`                                                      
  = note: required because of the requirements on the impl of `From<MidiError>` for `anyhow::Error`                    
  = note: required by `from`

...

error: aborting due to 6 previous errors

For more information about this error, try `rustc --explain E0277`.
error: could not compile `lp-control`

To learn more, run the command again with --verbose.
```

**Comment by kangalio** [comment]
Thank you for catching that

That must be because ConnectError stores the MidiInput/MidiOutput instance in them. I'll add some wrapper code that removes MidiInput/MidiOutput instances from ConnectError before storing them in MidiError

**Comment by kangalio** [comment]
Fixed in 64c55d7
